posthog/hogql/database/database.py

```diff
@@ -115,6 +125,29 @@ def create_hogql_database(team_id: int) -> Database:
     return database
 
 
+def determine_join_function(view):
+    def join_function(from_table: str, to_table: str, requested_fields: Dict[str, Any]):
+        from posthog.hogql import ast
+        from posthog.hogql.parser import parse_select
+
+        if not requested_fields:
+            raise HogQLException(f"No fields requested from {to_table}")
+
+        join_expr = ast.JoinExpr(table=parse_select(view.saved_query.query["query"]))
+        join_expr.join_type = "INNER JOIN"
+        join_expr.alias = to_table
+        join_expr.constraint = ast.JoinConstraint(
+            expr=ast.CompareOperation(
+                op=ast.CompareOperationOp.Eq,
+                left=ast.Field(chain=[from_table, view.from_join_key]),
+                right=ast.Field(chain=[to_table, view.to_join_key]),
+            )
+        )
```

Comment:
Nit: These can be just provided to the `ast.JoinExpr()` constructor, same as `table`